staging/src/k8s.io/component-base/metrics/metric.go

```diff
@@ -210,6 +212,33 @@ func (c *selfCollector) Collect(ch chan<- prometheus.Metric) {
 	ch <- c.metric
 }
 
+// metricWithExemplar is an interface that knows how to attach an exemplar to certain supported metric types.
+type metricWithExemplar interface {
+	withExemplar(v float64)
+}
+
+// exemplarMetric is a holds a context to extract exemplar labels from, and a metric to attach them to. It implements the metricWithExemplar interface.
+type exemplarMetric struct {
+	*Counter
+}
+
+// withExemplar attaches an exemplar to the metric.
+func (e *exemplarMetric) withExemplar(v float64) {
+	if m, ok := e.CounterMetric.(prometheus.ExemplarAdder); ok {
+		maybeSpanCtx := trace.SpanContextFromContext(e.ctx)
+		if maybeSpanCtx.IsValid() {
```

Comment:
```suggestion
		if maybeSpanCtx.IsValid() && maybeSpanCtx.IsSampled() {
```
Consider only recording exemplars for sampled spans.

Reply:
ACK, makes sense for exemplar metrics to be associated with sampled spans only.